### PR TITLE
DAOS-3105 control: Scrub engine environment

### PR DESCRIPTION
**Pull request by mjmac**
By default, all environment variables provided to the
engine subprocesses must be set by the control plane.

An optional per-engine config parameter (env_pass_through)
may be set to pass through variables set before the control
plane was started.

### CODE REVIEW

**Review by tanabarr — APPROVED** [review]
good that you decided this should be done, last time I approached it it was decided too risky. I hope it doesn't cause any subtle issues as it's a change that is enabled by default

**Comment by tanabarr** [comment]
also don't forget you have env_utils ScrubEnvironment* functions which might be able to be consolidated

**Comment by mjmac** [comment]
> 
> 
> also don't forget you have env_utils ScrubEnvironment* functions which might be able to be consolidated

Ha, totally forgot about those.

File: src/control/server/engine/exec_test.go [hunk 1]
```diff
@@ -119,13 +120,18 @@ func TestRunnerNormalExit(t *testing.T) {
 
 	// set this to control the behavior in TestMain()
 	os.Setenv(testModeVar, "RunnerNormalExit")
-	// verify that user env gets overridden by config
-	os.Setenv("OFI_INTERFACE", "bob0")
```
Comment (kjacque) [inline src/control/server/engine/exec_test.go:123]: Seems like this case could still be valid if the user allows passthrough for a variable that's also in the config. I think it's worth keeping that behavior defined in the tests.

**Pull request merged**